### PR TITLE
Chrono TrackVehicle toolkit, demo models

### PR DESCRIPTION
**Pull request by jcmadsen**

### CODE REVIEW

File: src/demos/irrlicht/CMakeLists.txt [hunk 1]
```diff
@@ -30,10 +31,11 @@ SET(DEMOS
   demo_aux_ref
   demo_rev_sph
   demo_spring
-  demo_ujoint
```
Comment (andrewseidl) [inline src/demos/irrlicht/CMakeLists.txt]: Was removing `demo_ujoint` intentional?

File: src/physics/ChSystem.cpp [hunk 2]
```diff
@@ -3541,19 +3563,19 @@ void ChSystem::ShowHierarchy(ChStreamOutAscii& m_file)
 	std::vector<ChBody*>::iterator ibody = Get_bodylist()->begin();
 	while (ibody != Get_bodylist()->end())
 	{
-		GetLog() << "     BODY:       " << (*ibody)->GetName() << "\n";
+		m_file << "     BODY:       " << (*ibody)->GetName() << "\n";
```
Comment (andrewseidl) [inline src/physics/ChSystem.cpp]: Good catch.

**Comment by andrewseidl** [comment]
LGTM in general.

- There is the issue of `demo_ujoint` getting lost in a merge.
- In the future you might want to squash your commits into more logical units.

**Comment by rserban** [comment]
Andrew, I sent some more comments/issues to Justin in a separate email.  Let's wait a bit before merging this pull request.   Also, given the huge number of commits in this PR, would it be better to directly add this to the main chrono repo?  What do other people think?

**Comment by andrewseidl** [comment]
> would it be better to directly add this to the main chrono repo?

I'd say the best idea is to squash the commits inside this PR into more logical units, with meaningful commit messages. Since PRs are currently our only code review tool, keep it in a PR/separate branch instead of committing to the main repo.

I'll start working on a `CONTRIBUTING.md` document to somewhat formalize the procedures.

**Comment by jcmadsen** [comment]
I've addressed comments from Radu and andrew at this point

**Comment by rserban** [comment]
> squash the commits inside this PR into more logical units

Andrew, how would one do that?  Will you explain this in the document you mentioned?

**Comment by rserban** [comment]
> I've addressed comments from Radu and andrew at this point

Justin, I mentioned to you that compilation fails with VS 2013 because you must include `<algorithm>` in sources where you use std::min.  I don't see that among your latest commits.  Did I miss something?

**Comment by jcmadsen** [comment]
just fixed that. It should work now.

**Comment by andrewseidl** [comment]
> Andrew, how would one do that?

[`git rebase -i`](https://www.atlassian.com/zh/git/tutorial/rewriting-git-history#!rebase-i)

> Will you explain this in the document you mentioned?

Yes. And once the document is created GitHub will display (a link to) it when someone opens an issue/PR.

**Comment by rserban** [comment]
The problem is that you **must** include `<algorithm>` in some sources (e.g. `DriveChain.cpp`, `LoopChain.cpp`, `TrackSoilBin.cpp`, `TrackVehicle.cpp`).

Instead, you removed its inclusion in `ChTrackVehicle.cpp`.

I thought you're using VS 2013.  Don't you?

**Comment by jcmadsen** [comment]
Radu and I decided that this should be merged into two sets of commits:
A) change to the existing Chrono source, and 
B) additions from the TrackVehicle library

**Pull request closed without merging**